Adding High Availability functionality to basic RDS deployment template

navesh123 opened the pull request:
This template is to add highly available functionality to three basic
RDS deployment templates. This template create additional Gateway and
Web server and add those servers to existing Gateway and web server farm
in RDS deployment.

azurecla commented:
Hi __@navesh123__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

azurecla commented:
__@navesh123__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

navesh123 commented:
Thanks for your comment.
Could you please let me know how to get the write access to this repository.

singhkay commented:
@navesh123 Please read through the contribution guide to make this template compatible to the repo guidelines https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE

Why do you need write access to the repo?

navesh123 commented:
@singhkay I dont need write access to the repo i have already deleted that request. 
I have added all the files as per the guidelines in contribution guide. Please let me know if i need to make more changes.

navesh123 commented:
@singhkay I have added all the files as per the guidelines in contribution guide. Please let me know if i need to make more changes.

gatneil commented on rds-deployment-HA/README.md:
```diff
@@ -0,0 +1,20 @@
+
+Click the button below to deploy
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fnavesh123%2Fazure-quickstart-templates%2Fmaster%2Frds-deployment-HA%2Fazuredeploy.json" target="_blank">
```
this is pointing to your repo (navesh/azure-quickstart-templates); should point to the Azure one (Azure/azure-quickstart-templates)

navesh123 replied:
Done changes to point it to  (Azure/azure-quickstart-templates)

gatneil commented on rds-deployment-HA/README.md:
```diff
@@ -0,0 +1,20 @@
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fnavesh123%2Fazure-quickstart-templates%2Fmaster%2Frds-deployment-HA%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Fnavesh123%2Fazure-quickstart-templates%2Fmaster%2Frds-deployment-HA%2Fazuredeploy.json" target="_blank">
```
same here

navesh123 replied:
Done changes to point it to  (Azure/azure-quickstart-templates)

gatneil commented on rds-deployment-HA/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,42 @@
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminPassword": {
+      "value": "Gen-Password"
```
should be all-caps

navesh123 replied:
Changed it to all-caps

singhkay commented on rds-deployment-HA/azuredeploy.json:
```diff
@@ -0,0 +1,397 @@
+        "description": "Unique public DNS prefix for the deployment. The fqdn will look something like '<dnsname>.westus.cloudapp.azure.com'. Up to 62 chars, digits or dashes, lowercase, should start with a letter: must conform to '^[a-z][a-z0-9-]{1,61}[a-z0-9]$'."
+      }
+    },
+    "loadBalancer": {
```
Is this an existing LB or a new one? The paramter name should reflect the intent for the user

navesh123 replied:
This is an existing loadbalancer which is deployed through RDS basic template. Information regarding this is already published in README file. The idea of building this resource is identical to primary deployment. The loadbalancer is defined in this template because the VM creation in this template is dependent on loadbalancer.

singhkay commented on rds-deployment-HA/azuredeploy.json:
```diff
@@ -0,0 +1,397 @@
+      },
+      "defaultValue": "loadBalancer"
+    },
+    "backendAddressPools": {
```
Seems like there is a template that needs to be deployed before deploying this template. Readme should clarify this

navesh123 replied:
We have added the README file accordingly. Please let us know if you want us to make any changes to README file.

singhkay commented on rds-deployment-HA/azuredeploy.json:
```diff
@@ -0,0 +1,397 @@
+        "description": "Number of RDSH instances"
+      }
+    },
+    "BrokerServer": {
```
camelCase on parameters

navesh123 replied:
Changes made as per the comment

singhkay commented on rds-deployment-HA/azuredeploy.json:
```diff
@@ -0,0 +1,397 @@
+        "description": "FQDN for Broker Server"
+      }
+    },
+    "WebURL": {
```
parameter name should clearer. What is this webURL of?

navesh123 replied:
We have updated the description comment for this. WebURL is simple public url name for RD gateway external FQDN.

singhkay commented on rds-deployment-HA/azuredeploy.json:
```diff
@@ -0,0 +1,397 @@
+        "description": "Netbios Name for Domain"
+      }
+    },
+    "storageAccountName": {
```
See #944 for storage account best practices. Can this be auto-generated?

navesh123 replied:
This template is not generating a new Storage account. This storage account name shall be picked from existing basic RDS deployment. Related Information is already updated in README file.

singhkay commented on rds-deployment-HA/azuredeploy.json:
File uris should be this repo

singhkay commented on rds-deployment-HA/azuredeploy.json:
since this commandtoexecute has a password, this commandtoexecute needs to be sent as part of protected settings

singhkay commented on rds-deployment-HA/metadata.json:
```diff
@@ -0,0 +1,7 @@
+  "itemDisplayName": "RD Session Collection High availability deployment",
+  "description": "This template provide high availability to RDWA and RDG servers in an existing RDSH deployment",
+  "summary": "Provide HA to RDWA and RDG in an existing RDSH deployment",
+  "githubUsername": "ashishsharma303,navesh123",
```
Can only be one author otherwise it's not going to index on Azure.com

navesh123 replied:
Changed has been made to metadata file as per he comments.

navesh123 commented on rds-deployment-HA/azuredeploy.json:
Changes made as per the comment. uris is from this repo

navesh123 commented on rds-deployment-HA/azuredeploy.json:
Changes made as per the comment. commandtoexecute has been moved to protected settings

singhkay commented on rds-deployment-HA/README.md:
```diff
@@ -1,13 +1,24 @@
+•	Post configurations for web/Gateway VM's such as defining the Machine keys for IIS modules.
+
+Prerequisites:
+RDS-deployment-HA is an extension to the Basic-RDS-Deployment and it is mandatory to deploy any one of the template as prerequisite “rds-deployment”, “rds-deployment-custom-image-rdsh”, “rds-deployment-existing-ad”
```
It's best to link these templates here so the user doesn't have to go find these. Please either link to Github location or Azure.com location.

singhkay commented:
@navesh123 Please see https://github.com/Azure/azure-quickstart-templates/pull/1896/files#r63797055 and fix in another PR. Will merge this in the meantime.
